### PR TITLE
retro-gtk: init at 1.0.2

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Motivation for this change

First step to get [Gnome Games](https://gitlab.gnome.org/GNOME/gnome-games) which has been asked in #68863.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 124168` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>retro-gtk</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124168` at 48cbbdad run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>retro-gtk</li>
  </ul>
</details>
<details>
  <summary>6 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

cmake is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/retro-gtk/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

pkg-config is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/retro-gtk/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

meson is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/retro-gtk/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

gtk-doc is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/retro-gtk/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

vala is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/retro-gtk/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

ninja is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/libraries/retro-gtk/default.nix:27:3:
```
   |
27 |   buildInputs = [
   |   ^
```
</li>
  </ul>
</details>

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 124168` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>retro-gtk</li>
  </ul>
</details>

File: pkgs/development/libraries/retro-gtk/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  pname = "retro-gtk";
+  version = "1.0.2";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/retro-gtk/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
+    sha256 = "1lnb7dwcj3lrrvdzd85dxwrlid28xf4qdbrgfjyg1wn1z6sv063i";
+  };
+
+  nativeBuildInputs = [
+    meson cmake pkg-config epoxy glib gtk3 libpulseaudio libsamplerate gobject-introspection vala gtk-doc ninja
```
Comment (jtojnar) [inline pkgs/development/libraries/retro-gtk/default.nix:28]: Build tools should go to `nativeBuildInputs`. Runtime dependencies to `buildInputs`.

https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md

Comment (DamienCassou) [reply pkgs/development/libraries/retro-gtk/default.nix]: Fixed!

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 124168` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>retro-gtk</li>
  </ul>
</details>

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 124168` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>retro-gtk</li>
  </ul>
</details>

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 124168` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>retro-gtk</li>
  </ul>
</details>

File: pkgs/development/libraries/retro-gtk/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  pname = "retro-gtk";
+  version = "1.0.2";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/retro-gtk/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
+    sha256 = "1lnb7dwcj3lrrvdzd85dxwrlid28xf4qdbrgfjyg1wn1z6sv063i";
+  };
+
+  nativeBuildInputs = [
+    meson cmake pkg-config vala gtk-doc ninja
```
[the remark this answers is not in the document]
Comment (DamienCassou) [reply pkgs/development/libraries/retro-gtk/default.nix]: you are right, it wasn't

[the remark this answers is not in the document]
Comment (DamienCassou) [reply pkgs/development/libraries/retro-gtk/default.nix]: done

File: pkgs/development/libraries/retro-gtk/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  ];
+
+  buildInputs = [
+     epoxy glib gtk3 libpulseaudio libsamplerate gobject-introspection
```
[the remark this answers is not in the document]
Comment (DamienCassou) [reply pkgs/development/libraries/retro-gtk/default.nix]: fixed

[the remark this answers is not in the document]
Comment (DamienCassou) [reply pkgs/development/libraries/retro-gtk/default.nix]: if I don't add it, I get:

```
Configuring retro-gtk-version.h using configuration
Did not find CMake 'cmake'
Found CMake: NO
Build-time dependency gobject-introspection-1.0 found: NO (tried pkgconfig)

retro-gtk/meson.build:142:2: ERROR: Dependency "gobject-introspection-1.0" not found, tried pkgconfig
```

So I moved it to `nativeBuildInputs`.

File: pkgs/development/libraries/retro-gtk/default.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  ];
+
+  patches = [
```
[the remark this answers is not in the document]
Comment (DamienCassou) [reply pkgs/development/libraries/retro-gtk/default.nix]: that makes sense, fixed

**Comment by jtojnar** [comment]
The code looks good to me now, did not test.

**Pull request merged**